### ISSUE TITLE
"pool_slots" may be None

### ISSUE THREAD

**Issue by suiting-young** [apache/airflow#15025]
https://github.com/apache/airflow/blob/b1ce429fee450aef69a813774bf5d3404d50f4a5/airflow/jobs/scheduler_job.py#L1041

This would trigger error
```
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```
as the `task_instance` is fetched directly (*without the patch of `refresh_from_db`*)
https://github.com/apache/airflow/blob/b1ce429fee450aef69a813774bf5d3404d50f4a5/airflow/jobs/scheduler_job.py#L930-L935

Related to #8153.

**Comment by XD-DENG** [comment]
Hi @suiting-young , may you please share the specific steps reproducing the error?

**Comment by suiting-young** [comment]
@XD-DENG after upgrade my instance from 1.10.14 to 2.0.1, some very old `TI`-s were fetched by scheduler.
I think those `TI` were **interrupted** (_not in finished state_) for some reason long ago, and _ideally_ already expired.
Maybe the new scheduler changed some rules, as those data didn't trigger problem on 1.10.14 (which already has `pool_slots` property).

**Comment by kaxil** [comment]
This was fixed in https://github.com/apache/airflow/pull/14406 and will be available as part of 2.0.2

**Comment by kaxil** [comment]
Workaround is to run:

```sql
UPDATE task_instance SET pool_slots = 1 WHERE pool_slots IS NULL;
```
